mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -776,7 +776,7 @@ private[ml] object RandomForest extends Logging {
               val categoryStats =
                 binAggregates.getImpurityCalculator(nodeFeatureOffset, featureValue)
               val centroid = if (categoryStats.count != 0) {
-                categoryStats.predict
+                categoryStats.calculate()
```

Comment:
This should use the soft prediction, not simply the impurity.  For regression, that means using predict(), but for binary classification, that will require using categoryStats.stats(1).